core/tests/protractor/suggestions.js

```diff
@@ -0,0 +1,87 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end tests for suggestions on explorations
+ */
+
+var dashboard = require('../protractor_utils/dashboard.js');
+var editor = require('../protractor_utils/editor.js');
+var general = require('../protractor_utils/general.js');
+var forms = require('../protractor_utils/forms.js');
+var player = require('../protractor_utils/player.js');
+var library = require('../protractor_utils/library.js');
+var users = require('../protractor_utils/users.js');
+var workflow = require('../protractor_utils/workflow.js');
+
+describe('Suggestions on Explorations', function() {
+  var EXPLORATION_TITLE = 'Sample Exploration';
+  var EXPLORATION_CATEGORY = 'Random';
+  var EXPLORATION_OBJECTIVE = 'To explore something new';
+  var EXPLORATION_LANGUAGE = 'English';
+
+  beforeEach(function() {
+    users.createUser('user1@ExplorationSuggestions.com',
+                     'authorExplorationSuggestions');
+    users.createUser('user2@ExplorationSuggestions.com',
+                     'suggesterExplorationSuggestions');
+    users.createUser('user3@ExplorationSuggestions.com',
+                     'studentExplorationSuggestions');
+  });
+
+  it('accepts a suggestion on a published exploration', function() {
+    users.login('user1@ExplorationSuggestions.com');
+    workflow.createAndPublishExploration(EXPLORATION_TITLE,
+                                         EXPLORATION_CATEGORY,
+                                         EXPLORATION_OBJECTIVE,
+                                         EXPLORATION_LANGUAGE);
+    browser.get(general.SERVER_URL_PREFIX);
+    users.logout();
+
+    // Suggester plays the exploration and suggests a change
+    users.login('user2@ExplorationSuggestions.com');
+    browser.get(general.LIBRARY_URL_SUFFIX);
+    library.playExploration(EXPLORATION_TITLE);
+
+    var suggestion = 'New Exploration';
+    var suggestionDescription = 'Uppercased the first letter';
+
+    player.submitSuggestion(suggestion, suggestionDescription);
+    users.logout();
+
+    // Exploration author reviews the suggestion and accepts it
+    users.login('user1@ExplorationSuggestions.com');
+    browser.get(general.SERVER_URL_PREFIX);
+    dashboard.navigateToExplorationEditor();
+    editor.getSuggestionThreads().then(function(threads) {
+      expect(threads.length).toEqual(1);
+      expect(threads[0]).toMatch(suggestionDescription);
+    });
+    editor.acceptSuggestion(suggestionDescription);
+    editor.navigateToPreviewTab();
+    editor.expectContentToMatch(forms.toRichText(suggestion));
+    users.logout();
+
+    // Student logs in and plays the exploration, finds the updated content
+    users.login('user3@ExplorationSuggestions.com');
+    browser.get(general.LIBRARY_URL_SUFFIX);
+    library.playExploration(EXPLORATION_TITLE);
+    player.expectContentToMatch(forms.toRichText(suggestion));
```

Comment:
This expectation fails but I have no idea why. I tried adding a wait before this line, but didn't help. When I used `browser.debugger()` to inspect the content, the suggested change was present in the exploration but in the test the content ends up to be an empty string. What am I missing?

@jacobdavis11 Can you have a look?

/cc @seanlip